### DISCUSSION TITLE
Realtime Quotas and Postgres Changes

### DISCUSSION THREAD

**Discussion by gangaramstyle** [supabase/supabase#27844] [Questions] [answered]
Hi! I had a quick question about how realtime messages get counted per Postgres change.

Let's say I have a table with the columns **row_id** and **payload** and there are two rows within the table:
[row_1, payload_A]
[row_2, payload_B]

If I have a user that is subscribed to changes in the table with a filter 'row_id=eq.row_1' and row_2 gets updated, would that still count as a realtime message? I know that the filter means that the client will not have an event fire, but I just want to make sure that the messages that are filtered out wouldn't still be counted against the realtime message quota. 

I'm assuming that the .on(...) function filters apply on the Supabase service side so that the filtered out messages are never 'created' in the first place, but figured I would double check before going down a path that might not scale at all.

Please let me know if anything is unclear! Thank you for any help!

**Comment by GaryAustin1** [comment, marked as answer]
That is my understanding.  The filter is sent to the realtime server on the websocket.  I've not watched the websocket in detail, but I don't ever recall seeing a message for a filtered table that should not be there, when I have.

**Reply by gangaramstyle** [reply]
Got it, that's good enough for me! :)

Thanks!
